src/Kernel/BlockClosure.class.st

```diff
@@ -336,13 +336,13 @@ BlockClosure >> ifError: errorHandlerBlock [
 	"Examples:
 		[1 whatsUpDoc] ifError: [:err :rcvr | 'huh?'].
 		[1 / 0] ifError: [:err :rcvr |
-			'ZeroDivide' = err
+			ZeroDivide = err class
 				ifTrue: [Float infinity]
 				ifFalse: [self error: err]]
 "
 
 	^ self on: Error do: [:ex |
-		errorHandlerBlock cull: ex description cull: ex receiver]
```

Comment:
I would also remove second parameter. It is incompatible change anyway. So it should not add more problems.
(my idea was to make ifError consistent to on:do:)

Reply:
Attempt in the commit kamk/pharo@415dfdefea0d5da0a49f7da62dff382687248ef7.